capstone/src/main/java/com/google/sps/servlets/StudentServlet.java

```diff
@@ -43,19 +85,44 @@ private static String convertToJsonUsingGson(StudentInfo info) {
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
     // TODO: All commented steps below
-    // 1. Get student object based on the logged in email
-    // 2. Add club to logged in student's club list if requested
-    // 3. Remove club from logged in student's club list if requested
-    // 4. Update student information with edited content
+    // 1. Add club to logged in student's club list if requested
+    // 2. Remove club from logged in student's club list if requested
+    // 3. Update student information with edited content
+
+    // Get student object based on the logged in email
+    UserService userService = UserServiceFactory.getUserService();
+    String userEmail = userService.getCurrentUser().getEmail();
+
+    // Add user to Datastore if not already stored
+    Query query = new Query(userEmail);
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    PreparedQuery results = datastore.prepare(query);
+
+    if (Streams.stream(results.asIterable()).count() == 0) {
```

Comment:
nit: could you just check if results.asIterable() is empty? Not sure if you need to turn it into a stream to check that?

Reply:
Done! I used Guava's Iterables to check if it was empty instead.